### PR TITLE
Add initial version of enrich processor docs.

### PR DESCRIPTION
**Pull request by martijnvg**
Relates to #32789

### CODE REVIEW

File: docs/reference/ingest/processors/enrich.asciidoc
Comment (jakelandis) [inline docs/reference/ingest/processors/enrich.asciidoc]: this feels that this should live outside of the processor doc. 
Perhaps a top level item "Enriching data" in Ingest heading https://www.elastic.co/guide/en/elasticsearch/reference/current/ingest.html ?

File: docs/reference/ingest/processors/enrich.asciidoc
Comment (hub-cap) [inline docs/reference/ingest/processors/enrich.asciidoc]: `data<two spaces>in the user's`

File: docs/reference/ingest/processors/enrich.asciidoc
Comment (hub-cap) [reply docs/reference/ingest/processors/enrich.asciidoc]: ++ to this, like how `Conditional Execution in Pipelines` has its own call out / section, right @jakelandis 	?

File: docs/reference/ingest/processors/enrich.asciidoc
Comment (hub-cap) [inline docs/reference/ingest/processors/enrich.asciidoc]: who this? should we put an email address that does not (potentially) exist?

**Review by hub-cap — APPROVED** [review]
nice work. You may want to put something in the docs that says "if you modify your enrich index, you need to re-execute the index. Also, please dont muck w/ the backing indices that are created!!!"

And i like jakes suggestion about moving most of this up.

File: docs/reference/ingest/processors/enrich.asciidoc
Comment (martijnvg) [reply docs/reference/ingest/processors/enrich.asciidoc]: Do you mean only the `Enrich Policy Definition` section or also the getting started and description of APIs?

Comment (jakelandis) [reply docs/reference/ingest/processors/enrich.asciidoc]: yes, very much like `Conditional Execution in Pipelines` where there can multiple sub pages to describe the Enriching process. The processor definition would still exist, but would be a pretty boring reference. 

This PR doesn't need to include the entirety of the doc, but I think having a separate top level where one sub item can be the policy definition doc would be a nice way to present this data.

Comment (jakelandis) [reply docs/reference/ingest/processors/enrich.asciidoc]: so to actually answer your question... the processor reference doc would go with the processors, the getting started, policy, and API definitions would all be sub headers under `Enriching ingest data` (or something like that).

Comment (martijnvg) [reply docs/reference/ingest/processors/enrich.asciidoc]: I've pushed: https://github.com/elastic/elasticsearch/pull/45084/commits/9ed37ece49e39d3d60cc5c7689d1c9942a3c0efe

**Review by jakelandis — APPROVED** [review]
LGTM assuming doc build passes

also, some of the content will be changing based on discussions earlier today and will want to re-review the full docs once those changes are landed.

**Pull request merged**
